lib/AST/AccessRequests.cpp

```diff
@@ -40,6 +41,9 @@ AccessLevel
 AccessLevelRequest::evaluate(Evaluator &evaluator, ValueDecl *D) const {
   assert(!D->hasAccess());
 
+  // Access notes may add attributes that affect this calculus.
+  (void)evaluateOrDefault(evaluator, ApplyAccessNoteRequest{D}, {});
```

Comment:
ApplyAccessNoteRequest also introduces DynamicAttr, UsableFromInlineAttr and ObjCAttr so doing it here doesn't make any sense as there's no guarantee that someone won't query those attributes before checking the access level.

Reply:
Specifically, what you need to do is the following:
- change the places that directly check for UsableFromInlineAttr to call a method on ValueDecl which evaluates a new UsableFromInlineRequest
- split up ApplyAccessNoteRequest
- in AccessLevelRequest, apply the access level from the API note
- in DynamicRequest, apply the dynamic attribute from the API note
- in ObjCRequest, apply the objc attribute from the API note
- in your new UsableFromInlineRequest, apply the usableFromInline attribute from the API note